client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,286 @@
+				},
+			},
+			{
+				id: 'update',
+				label: translate( 'Update available' ),
+				getValue: ( { item }: { item: Plugin } ) => {
+					// Used exclusively for sorting
+					return item.status?.includes( PLUGINS_STATUS.UPDATE ) ? 'a' : 'b';
+				},
+				enableHiding: false,
+				render: ( { item }: { item: Plugin } ) => {
+					if ( item.status?.includes( PLUGINS_STATUS.UPDATE ) ) {
+						return (
+							<Button
+								variant="secondary"
+								onClick={ () => bulkActionDialog( PluginActions.UPDATE, [ item ] ) }
+							>
+								{ translate( 'Update to version %(version)s', {
+									args: {
+										version: item?.update?.new_version || '',
+									},
+								} ) }
+							</Button>
+						);
+					}
+				},
+			},
+		],
+		[]
+	);
+
+	const actions = [
+		{
+			id: 'manage-plugin',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				plugins.length && navigate( '/plugins/' + plugins[ 0 ].slug );
+			},
+			label: translate( 'Manage Plugin' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: false,
+		},
+		{
+			id: 'activate-plugin',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				bulkActionDialog( PluginActions.ACTIVATE, plugins );
+			},
+			label: translate( 'Activate' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: true,
+			icon: <Icon icon={ link } />,
+		},
+		{
+			id: 'deactivate-plugin',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				bulkActionDialog( PluginActions.DEACTIVATE, plugins );
+			},
+			label: translate( 'Deactivate' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: true,
+			icon: <Icon icon={ linkOff } />,
+		},
+		{
+			id: 'enable-autoupdate',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				bulkActionDialog( PluginActions.ENABLE_AUTOUPDATES, plugins );
+			},
+			label: translate( 'Enable Autoupdate' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: true,
+		},
+		{
+			id: 'disable-autoupdate',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				bulkActionDialog( PluginActions.DISABLE_AUTOUPDATES, plugins );
+			},
+			label: translate( 'Disable Autoupdate' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: true,
+		},
+		{
+			id: 'remove-plugin',
+			href: `some-url`,
+			callback: ( plugins: Array< Plugin > ) => {
+				bulkActionDialog( PluginActions.REMOVE, plugins );
+			},
+			label: translate( 'Remove' ),
+			isExternalLink: true,
+			isEnabled: true,
+			supportsBulk: true,
+			icon: <Icon icon={ trash } />,
+		},
+	];
+
+	// Set initial state for the data views
+	const pluginsPerPage = 15;
+	initialDataViewsState.perPage = pluginsPerPage;
+	initialDataViewsState.search = initialSearch;
+	initialDataViewsState.fields = [ 'plugins', 'sites', 'update' ];
+	const [ dataViewsState, setDataViewsState ] = useState( initialDataViewsState );
```

Comment:
We should never mutate objects like this. Let's create a new copy. 

Feel free to do it in lazy init, so it happens only initially, like this (untested):

```suggestion
	const [ dataViewsState, setDataViewsState ] = useState( () => ( {
		...initialDataViewsState,
		perPage: 15,
		search: initialSearch,
		fields: [ 'plugins', 'sites', 'update' ],
	} ) );
```

Reply:
Good idea, TIL about lazy init. Thanks for the suggestion, [I applied it here](https://github.com/Automattic/wp-calypso/pull/95561/commits/47b464ca38bd2c1e744ee8ad4b93e8636c36c895).